src/shared/hooks/useMediaUpload/useMediaUpload.test.ts

```diff
@@ -0,0 +1,94 @@
+import { renderHook, act } from '@testing-library/react';
+import axios from 'axios';
+
+import { useMediaUpload } from './useMediaUpload';
+import { UseMediaUploadReturn } from './useMediaUpload.types';
+
+jest.mock('axios');
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+const mockGetMediaUploadUrl = jest.fn();
+jest.mock('shared/hooks/useAsync', () => ({
+  useAsync: () => ({ execute: mockGetMediaUploadUrl }),
+}));
+const mockCallback = jest.fn();
+const mockErrorCallback = jest.fn();
+const mockFinallyCallback = jest.fn();
+const url = 'https://example.com/image.jpg';
+const uploadUrl = 'https://example.com/upload';
+const file = new File(['file contents'], 'test.jpg', { type: 'image/jpeg' });
+const fileName = 'test.jpg';
+const uploadUrlResponse = {
+  data: {
+    result: {
+      url,
+      uploadUrl,
+      fields: {
+        key: 'test-key',
+        policy: 'test-policy',
+        signature: 'test-signature',
+        'x-amz-security-token': 'test-token',
+      },
+    },
+  },
+};
+const useMediaUploadProps = {
+  callback: mockCallback,
+  errorCallback: mockErrorCallback,
+  finallyCallback: mockFinallyCallback,
+};
+
+const testUploadFlow = async (result: { current: UseMediaUploadReturn }) => {
+  act(() => {
+    result.current.executeMediaUpload({ file, fileName });
+  });
+
+  expect(result.current.isLoading).toBe(true);
+
+  // wait for the next update
+  await act(async () => {
```

Comment:
try to use `rerender` instead

Reply:
Using `rerender` doesn't achieve the same result in this context. The test requires waiting for the state of the hook to be updated, which is not effectively triggered by `rerender` alone.